### PR TITLE
[WIP][TEST][test-maven] createSparkSession should be synchronized

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
This is a test to verify the reason of recent consequtive failures on Spark master branch.
- https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala [hunk 1]
```diff
@@ -61,8 +61,10 @@ trait SharedSQLContext extends SQLTestUtils with BeforeAndAfterEach with Eventua
    */
   protected override def beforeAll(): Unit = {
     SparkSession.sqlListener.set(null)
-    if (_spark == null) {
-      _spark = createSparkSession
+    synchronized {
```
Comment (srowen) [inline sql/core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala]: `beforeAll` ought to be called by one thread in the test framework before any tests start. I doubt this is the reason, but, maybe I don't have the full context here.

**Comment by dongjoon-hyun** [comment]
Thank you for review, @srowen . You're right.

**Comment by dongjoon-hyun** [comment]
Ur, @srowen .
The contract seems to be just [create new instance of the suite class](https://github.com/scalatest/scalatest/blob/3.1.x/scalatest/src/main/scala/org/scalatest/OneInstancePerTest.scala#L21-L23).

> Trait that facilitates a style of testing in which each test is run in its own instance
> of the suite class to isolate each test from the side effects of the other tests in the
> suite.

I'll just retrigger this PR to see the result simply.

**Comment by srowen** [comment]
@dongjoon-hyun I don't see that trait applied here, is it inherited from something else? I was focused on `BeforeAndAfterEach`. However if each test runs in its own instance, then there should also be no synchronization issue? because the same instances aren't being modified

**Comment by dongjoon-hyun** [comment]
Yep. It's my mistake. It was not.

**Pull request closed without merging**